libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -606,18 +684,53 @@ bool AP_DDS_Client::create()
             requests[2] = dreader_req_id;
 
             if (!uxr_run_session_until_all_status(&session, requestTimeoutMs, requests, status, nRequests)) {
-                GCS_SEND_TEXT(MAV_SEVERITY_ERROR,"XRCE Client: Topic/Sub/Reader session request failure for index '%d'",(int)i);
+                debug("XRCE Client: Topic/Sub/Reader session request failure for index '%u'",i);
                 for (uint8_t s = 0 ; s < nRequests; s++) {
-                    GCS_SEND_TEXT(MAV_SEVERITY_ERROR,"XRCE Client: Status '%d' result '%u'", s, status[s]);
+                    debug("XRCE Client: Status '%d' result '%u'", s, status[s]);
                 }
                 // TODO add a failure log message sharing the status results
                 return false;
             } else {
-                GCS_SEND_TEXT(MAV_SEVERITY_INFO,"XRCE Client: Topic/Sub/Reader session pass for index '%d'",(int)i);
+                debug("XRCE Client: Topic/Sub/Reader session pass for index '%u'",i);
                 uxr_buffer_request_data(&session, reliable_out, topics[i].dr_id, reliable_in, &delivery_control);
             }
         }
     }
+
+    // ROS-2 Service : else case for service requests
+
+    for (uint16_t i = 0; i < ARRAY_SIZE(services); i++) {
+
+        //status requests
+        uint16_t request;
```

Comment:
This has larger scope than needed. It's just used to create a copy of data once. Can you try deleting it?

Reply:
Resolved